code-studio/src/js/progress.js

```diff
@@ -191,32 +141,33 @@ progress.bestResultLevelId = function (levelIds, serverProgress, clientProgress)
   return bestId;
 };
 
-function loadProgress(scriptData, currentLevelId) {
+function loadProgress(scriptData, currentLevelId, saveAnswersBeforeNavigation = false) {
 
   let store = createStore((state = [], action) => {
```

Comment:
The default state in your reducer is empty array, but you seem to treat it as an object throughout.  Is this a mistake?